geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/AlterMappingCommand.java

```diff
@@ -81,49 +77,44 @@ public ResultModel alterMapping(
           specifiedDefaultValue = "") String[] fieldMappings) {
     // input
     Set<DistributedMember> targetMembers = getMembers(null, null);
-    ConnectorService.RegionMapping newMapping = new ConnectorService.RegionMapping(regionName,
+    RegionMapping newMapping = new RegionMapping(regionName,
         pdxClassName, table, connectionName, keyInValue);
     newMapping.setFieldMapping(fieldMappings);
 
-    ConfigurationPersistenceService ccService = getConfigurationPersistenceService();
-    // if cc is running, you can only alter connection available in cc service.
-    if (ccService != null) {
-      // search for the connection that has this id to see if it exists
-      CacheConfig cacheConfig = ccService.getCacheConfig(CLUSTER_CONFIG);
-      ConnectorService service =
-          cacheConfig.findCustomCacheElement("connector-service", ConnectorService.class);
-      if (service == null) {
-        throw new EntityNotFoundException("mapping with name '" + regionName + "' does not exist.");
-      }
-      ConnectorService.RegionMapping mapping =
-          CacheElement.findElement(service.getRegionMapping(), regionName);
-      if (mapping == null) {
-        throw new EntityNotFoundException("mapping with name '" + regionName + "' does not exist.");
-      }
-    }
-
     // action
     List<CliFunctionResult> results =
         executeAndGetFunctionResult(new AlterMappingFunction(), newMapping, targetMembers);
     ResultModel result =
         ResultModel.createMemberStatusResult(results, EXPERIMENTAL, null, false, true);
-
+    if (!result.isSuccessful()) {
+      return result;
+    }
     // find the merged regionMapping from the function result
     CliFunctionResult successResult =
         results.stream().filter(CliFunctionResult::isSuccessful).findAny().get();
-    ConnectorService.RegionMapping mergedMapping =
-        (ConnectorService.RegionMapping) successResult.getResultObject();
+    RegionMapping mergedMapping =
+        (RegionMapping) successResult.getResultObject();
     result.setConfigObject(mergedMapping);
     return result;
   }
 
   @Override
-  public void updateClusterConfig(String group, CacheConfig config, Object element) {
-    ConnectorService.RegionMapping mapping = (ConnectorService.RegionMapping) element;
-    ConnectorService service =
-        config.findCustomCacheElement("connector-service", ConnectorService.class);
-    // service is not nul at this point
-    CacheElement.removeElement(service.getRegionMapping(), mapping.getId());
-    service.getRegionMapping().add(mapping);
+  public void updateClusterConfig(String group, CacheConfig cacheConfig, Object element) {
+    RegionMapping newCacheElement = (RegionMapping) element;
+    RegionMapping existingCacheElement = cacheConfig.findCustomRegionElement(
+        newCacheElement.getRegionName(), newCacheElement.getId(), RegionMapping.class);
+
+    cacheConfig
```

Comment:
Can this be replaced with single forEach
.forEach(regionConfig -> {
  regionConfig.getCustomRegionElements().remove(existingCacheElement));
  regionConfig.getCustomRegionElements().add(newCacheElement))
}

It will be clear to see the element is removed and new one is added.

Reply:
This command is going away